.github/workflows/publish.yml

```diff
@@ -0,0 +1,48 @@
+name: Publish package to PyPI
+
+on:
+  push:
+    branches: [main]
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yml
+
+  publish:
+    name: Publish distribution to PyPI
+    runs-on: ubuntu-latest
+    environment: pypi_prod
+    needs: [build]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
```

Comment:
Please checkout @baywet question on using 3.12 at https://github.com/microsoft/kiota-serialization-json-python/pull/211#discussion_r1446088668. Any reason not to?

Reply:
Switched to 3.12